Source/WebKit/webpushd/_WKMockUserNotificationCenter.mm

```diff
@@ -31,14 +31,18 @@
 
 #if HAVE(FULL_FEATURED_USER_NOTIFICATIONS)
 
-static NSMutableArray *notificationsByBundleIdentifier(NSString *bundleIdentifier)
+@interface _WKMockUserNotificationCenter ()
+- (instancetype)_internalInitWithBundleIdentifier:(NSString *)bundleIdentifier;
+@end
+
+static _WKMockUserNotificationCenter *centersByBundleIdentifier(NSString *bundleIdentifier)
 {
-    static NeverDestroyed<RetainPtr<NSMutableDictionary>> notifications = adoptNS([NSMutableDictionary new]);
+    static NeverDestroyed<RetainPtr<NSMutableDictionary>> centers = adoptNS([NSMutableDictionary new]);
 
-    if (!notifications->get()[bundleIdentifier])
-        notifications->get()[bundleIdentifier] = [NSMutableArray arrayWithCapacity:1];
+    if (!centers->get()[bundleIdentifier])
+        centers->get()[bundleIdentifier] = [[_WKMockUserNotificationCenter alloc] _internalInitWithBundleIdentifier:bundleIdentifier];
 
-    return notifications->get()[bundleIdentifier];
+    return centers->get()[bundleIdentifier];
 }
 
 static NSMutableSet *notificationPermissions()
```

Comment:
I guess in a future patch we should get rid of this and move the boolean state to `_WKMockUserNotificationCenter`, now that we're making one center per bundle ID.